kolibri/core/kolibri_plugin.py

```diff
@@ -3,49 +3,28 @@
 """
 from __future__ import absolute_import, print_function, unicode_literals
 
-from django.core.urlresolvers import reverse
-from django.utils.translation import ugettext as _
-from kolibri.plugins.base import KolibriFrontEndPluginBase, KolibriPluginBase
-from kolibri.plugins.hooks import FRONTEND_PLUGINS, NAVIGATION_POPULATE
+from django.core.urlresolvers import reverse_lazy
+from django.utils.translation import ugettext_lazy as _
+from kolibri.core.webpack.hooks import FrontEndCoreHook
+from kolibri.plugins.base import KolibriPluginBase
 
-
-def main_navigation():
-    """
-    Callback for the plugin below
-    :return: A list of nav menu items used in the NAVIGATION_POPULATE hook.
-    """
-    return [{
-        'menu_name': _("Start page"),
-        'menu_url': reverse('kolibri:index'),
-    }]
+from . import hooks
 
 
 class KolibriCore(KolibriPluginBase):
     """
     The most minimal plugin possible. Because it's in the core, it doesn't define ``enable`` or ``disable``. Those
     methods should never be called for this plugin.
     """
-    def hooks(self):
+    pass
 
-        return {
-            NAVIGATION_POPULATE: main_navigation
-        }
 
+class MainNavigationItem(hooks.NavigationHook):
+    label = _("Start page")
+    url = reverse_lazy('kolibri:index')
 
-class KolibriCoreFrontEnd(KolibriFrontEndPluginBase):
-    """
-    Plugin to handle
-    """
-    entry_file = "assets/src/kolibri_core_app.js"
-    external = True
-    core = True
-
-    def hooks(self):
-        return {
-            FRONTEND_PLUGINS: self._register_front_end_plugins
-        }
-
-PLUGINS = (
-    KolibriCore,
-    KolibriCoreFrontEnd,
-)
+
+class DefaultFrontEndAsset(FrontEndCoreHook):
+    unique_slug = "default_frontend"
+    src_file = "kolibri/core/assets/src/kolibri_core_app.js"
+    static_dir = "kolibri/core/static"
```

Comment:
I think this should be a computed property, based on the app that the hook is defined in - also, it should return an absolute path, so that a plugin that is outside of the kolibri project dir can also be built in this way too.